File: frontend/src/commands/environment-proxy-group.ts
import { invoke } from './index'

export interface ProxyGroup {
    name: string;                    // 分组名称
    description?: string;            // 分组描述
}
export const environment_proxy_group_query_id = async (id: number): Promise<any> => {
    return await invoke('environment_proxy_group_query_id', { id })
};

export const environment_proxy_group_query = async (pageNum: number, pageSize: number): Promise<any> => {
    return await invoke('environment_proxy_group_query', { pageNum, pageSize })
};

export const environment_proxy_group_create = async (payload: ProxyGroup): Promise<any> => {
    return await invoke('environment_proxy_group_create', { payload })
};

export const environment_proxy_group_modify = async (id: number, payload: ProxyGroup): Promise<any> => {
    return await invoke('environment_proxy_group_modify', { id, payload })
};


export const environment_proxy_group_delete = async (id: number): Promise<any> => {
    return await invoke('environment_proxy_group_delete', { id })
};

File: crates/lp-views/src/command/environment_proxy_group.rs
// 导入JSON响应类型
use lp_cores::requests::JsonRespnse;

// 根据ID查询代理分组
#[tauri::command]
pub async fn environment_proxy_group_query_id(id: u32) -> Result<JsonRespnse, tauri::Error> {
    Ok(lp_services_remote::requests::environment_proxy_group::query_by_id(id).await?)
}

// 分页查询代理分组列表
#[tauri::command]
pub async fn environment_proxy_group_query(
    page_num: u32,
    page_size: u32,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(lp_services_remote::requests::environment_proxy_group::query(page_num, page_size).await?)
}

// 创建新的代理分组
#[tauri::command]
pub async fn environment_proxy_group_create(
    payload: lp_models::environment_proxy_group::ProxyGroup,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(lp_services_remote::requests::environment_proxy_group::create(payload).await?)
}

// 修改代理分组信息
#[tauri::command]
pub async fn environment_proxy_group_modify(
    payload: lp_models::environment_proxy_group::ProxyGroup,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(lp_services_remote::requests::environment_proxy_group::modify(payload).await?)
}

// 删除代理分组
#[tauri::command]
pub async fn environment_proxy_group_delete(id: u32) -> Result<JsonRespnse, tauri::Error> {
    Ok(lp_services_remote::requests::environment_proxy_group::delete(id).await?)
}
